tests/unit/test_scraper.py

```
@@ -89,418 +220,159 @@
-        mock_get.assert_called_once()
-        assert prices != test_data  # 新しいデータが取得される
+def test_scrape_url_validation_error_handling(mock_config, mock_response):
+    """検証エラーのハンドリングテスト"""
+    mock_response.text = """
+    <div class="tr">
+        <h2>Invalid Model</h2>
+        <div class="td2wrap">Invalid Price</div>
+    </div>
+    """
+    with patch('requests.get', return_value=mock_response):
+        prices = scrape_url('https://example.com/url1', mock_config)
```

Ruff (0.8.2)

Undefined name `scrape_url`

(F821)